cmd/prepare-storage.go

```diff
@@ -238,8 +234,26 @@ func waitForFormatXL(ctx context.Context, firstDisk bool, endpoints EndpointList
 	retryTimerCh := newRetryTimerSimple(doneCh)
 	for {
 		select {
-		case _ = <-retryTimerCh:
-			format, err := connectLoadInitFormats(firstDisk, endpoints, setCount, disksPerSet)
+		case retry := <-retryTimerCh:
+			// Initialize all storage disks
+			storageDisks, err := initStorageDisks(endpoints)
+			if err != nil {
+				return nil, err
+			}
+			defer closeStorageDisks(storageDisks)
```

Comment:
Do not defer this increases stack size on many retries. If you look at the previous code. It specifically closed the storage disks before returning format. If you wish to log move it inside that function instead.

Reply:
> Do not defer this increases stack size on many retries. If you look at the previous code. It specifically closed the storage disks before returning format. If you wish to log move it inside that function instead.

Good catch!